### PR TITLE
Odr 547 master

### PR DESCRIPTION
**Pull request by jimturnquist**
master pull request for ODR-547 work.

### CODE REVIEW

File: lib/graphs/discovery-refresh-delayed-graph.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        },
+        {
+            label: 'generate-sku',
+            waitOn: {
+                'discovery-refresh-graph': 'succeeded'
+            },
+            taskName: 'Task.Catalog.GenerateSku',
+        },
+        {
+            label: 'generate-enclosure',
```
Comment (heckj) [inline lib/graphs/discovery-refresh-delayed-graph.js]: double checking - is this task idempotent, or will you get multiple enclosures by repeatedly running this?

Comment (heckj) [inline lib/graphs/discovery-refresh-delayed-graph.js]: (looks like if it fails if one already exists, we're safe with `ignoreFailure: true` just below)

**Comment by heckj** [comment]
Question about the enclosures (should have spotted that earlier), but otherwise :+1: if you're good with the ramifications. Side note to be clear, if the delayed graph is invoked, the workflow will be "in existence" until the node reboots and does that process, which means other workflows may not be able to be run against it - in particular, I don't know that you'll be able to invoke a reboot workflow against that same node...

**Comment by jimturnquist** [comment]
@heckj Yep.  The delayed case will work exactly as you mentioned.  I believe the use case is when a node is taken out of service and then added back in.  The node id should "stick" but be refreshed on reboot.

File: lib/graphs/discovery-refresh-delayed-graph.js (hunk 1 above)
Comment (iceiilin) [inline lib/graphs/discovery-refresh-delayed-graph.js]: If the enclosure relation information already exists, the job will finish successfully without changing database. So it is fine here

**Comment by jimturnquist** [comment]
@RackHD/corecommitters I noticed this was still hanging out there.  I wasn't sure if there was something being waited on in regards to this.  The release 1.1 version is already in.

**Comment by jlongever** [comment]
+1  
just waiting for some jenkin's CI cleanup wrt recent workflow engine commits.

**Comment by benbp** [comment]
:+1:

**Comment by benbp** [comment]
test this please

**Pull request merged**